samples/basic/parallel_join.py

```diff
@@ -14,38 +14,44 @@
 # limitations under the License.
 
 
-import kfp.dsl as dsl
+import kfp
+from kfp import dsl
+
+def gcs_download_op(url):
+    return dsl.ContainerOp(
+        name='GCS - Download',
+        image='google/cloud-sdk:216.0.0',
+        command=['sh', '-c'],
+        arguments=['gsutil cat $0 | tee $1', url, '/tmp/results.txt'],
```

Comment:
Could you explain how $0 and $1 get replaced with the url and 'tmp/..'?

Reply:
It's shell script, right?
The full command-line is 

```sh -c 'gsutil cat $0 | tee $1' url /tmp/results.txt```

`$0` and `$1` are the first and the second arguments to sh.

P.S. If we were running a shell script file, then you'd have to use `$1` and `$2`